misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java

```diff
@@ -81,14 +82,66 @@ public void runTest(String sourceRoot, Path[] sourceFilePaths, List<String> grou
         TesterinaRegistry.getInstance().setGroups(groups);
         TesterinaRegistry.getInstance().setShouldIncludeGroups(shouldIncludeGroups);
 
+        // Compile and build the test suites
+        compileAndBuildSuites(sourceRoot, sourceFilePaths);
+        // execute the test programs
+        execute();
+        // print the report
+        tReport.printSummary();
+    }
+
+    /**
+     * lists the groups available in tests.
+     *
+     * @param sourceRoot      source root of the project
+     * @param sourceFilePaths package or program file paths
+     * @return
+     */
+    public void listGroups(String sourceRoot, Path[] sourceFilePaths) {
+        //Build the test suites
+        compileAndBuildSuites(sourceRoot, sourceFilePaths);
+        List groupList = getGroupList();
+        outStream.println("Following groups are available : ");
+        outStream.println(groupList);
+    }
+
+    /**
+     * Returns a distinct list of groups in test functions.
+     *
+     * @return a list of groups
+     */
+    public List getGroupList() {
+
+        Map<String, TestSuite> testSuites = TesterinaRegistry.getInstance().getTestSuites();
+        if (testSuites.isEmpty()) {
+            throw new BallerinaException("No test functions found in the provided ballerina files.");
```

Comment:
Suggestion to change the error msg to "No tests found in the Ballerina project."